### PR TITLE
Set operation name for consumer spans

### PR DESCRIPTION
**Pull request by vishweshbankwar**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Customizations/Models/TelemetryItem.cs [hunk 1]
```diff
@@ -23,15 +23,30 @@ internal partial class TelemetryItem
...
+                else
+                {
+                    Tags[ContextTagKeys.AiOperationName.ToString()] = activity.DisplayName;
+                }
+
+                // Set ip in case of server spans only.
+                if (activity.Kind == ActivityKind.Server)
+                {
+                    var locationIp = TraceHelper.GetLocationIp(ref activityTagsProcessor.MappedTags);
+                    if (locationIp != null)
```
Comment (rajkumar-rangaraj) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Customizations/Models/TelemetryItem.cs:45]: Do we need this check or leave it for serializer to handle it?

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Customizations/Models/TelemetryItem.cs [hunk 2]
```diff
@@ -23,15 +23,30 @@ internal partial class TelemetryItem
...
+                {
+                    Tags[ContextTagKeys.AiOperationName.ToString()] = TraceHelper.GetOperationName(activity, ref activityTagsProcessor.MappedTags);
+                }
+                else
+                {
+                    Tags[ContextTagKeys.AiOperationName.ToString()] = activity.DisplayName;
+                }
+
+                // Set ip in case of server spans only.
+                if (activity.Kind == ActivityKind.Server)
```
[the remark this answers is not in the document]
Comment (rajkumar-rangaraj) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Customizations/Models/TelemetryItem.cs:42]: I see its already in it.

[the remark this answers is not in the document]
Comment (vishweshbankwar) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Customizations/Models/TelemetryItem.cs:42]: Not sure if I follow. The check is inside `activity.GetTelemetryType() == TelemetryType.Request` but we still need to check on kind as we don't want to set ip for `Consumer` spans which are also requests. Let me know if I misunderstood your suggestion.

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Customizations/Models/TelemetryItem.cs (hunk 1 above)
Comment (vishweshbankwar) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Customizations/Models/TelemetryItem.cs:45]: It serializes the value `"ai.location.ip":null`

**Review by rajkumar-rangaraj — APPROVED** [review]

**Pull request merged**